syntax = "proto3";

package user;

service UserService {
  rpc RegisterUser (RegisterRequest) returns (RegisterResponse) {}
  rpc AuthenticateUser (AuthRequest) returns (AuthResponse) {}
  rpc GetUser (UserRequest) returns (User) {}
}

message RegisterRequest {
  string username = 1;
  string password = 2;
  string email = 3;
}

message RegisterResponse {
  bool success = 1;
  string message = 2;
  int32 user_id = 3;
}

message AuthRequest {
  string username = 1;
  string password = 2;
}

message AuthResponse {
  bool success = 1;
  string message = 2;
  string token = 3;  // You might want to return a JWT token for session management
}

message UserRequest {
  int32 id = 1;
}

message User {
  int32 id = 1;
  string username = 2;
  string email = 3;
  string store_url = 4;
  string api_key = 5;
  string api_access_token = 6;
  string api_version = 7;
}